experimental/dds/tree2/src/test/feature-libraries/optional-field/optionalChangeRebaser.spec.ts

```diff
@@ -120,13 +135,15 @@ function invert(change: TaggedChange<OptionalChangeset>): OptionalChangeset {
 function rebase(
 	change: OptionalChangeset,
 	base: TaggedChange<OptionalChangeset>,
+	metadataArg?: RevisionMetadataSource,
 ): OptionalChangeset {
 	deepFreeze(change);
 	deepFreeze(base);
 
-	const metadata = defaultRevisionMetadataFromChanges([base, makeAnonChange(change)]);
+	const metadata =
+		metadataArg ?? defaultRevisionMetadataFromChanges([base, makeAnonChange(change)]);
 	const moveEffects = failCrossFieldManager;
-	const idAllocator = idAllocatorFromMaxId(getMaxId(change, base.change));
+	const idAllocator = idAllocatorFromMaxId((getMaxId(change, base.change) ?? 0) + 1);
```

Comment:
Why is this ` ?? 0) + 1` stuff needed?

Reply:
It could just be `const idAllocator = failIdAllocator;` now: at one point while working on this change the rebase logic for optional field did sometimes use the id allocator passed to it, so it was necessary to pass a legitimate one.

If you take for granted that optional field needs one here, I think the `?? 0) + 1` is reasonable: when rebasing a child-only change over another child-only change, neither change has any local ids, and getMaxId returns undefined there. +1 guarantees no reuse of ids. Admittedly I wasn't particularly careful about whether the id allocator returns the value pre-increment or post-increment, so maybe it's not necessary. but not allocating an id isn't a big deal :)

Reply:
I'm confused: if neither changes have IDs then how could there be any reuse of IDs?
Note that `idAllocatorFromMaxId`...
- accepts undefined (in which case the next ID it generates will be 0). 
- will next generate N+1 when seeded with N.

Reply:
I'm sure `idAllocatorFromMaxId(getMaxId(change, base.change))` is sufficient then. I wasn't aware of either property of `idAllocatorFromMaxId` when implementing and didn't want to bother checking for test code.